example/pico-ssram/firmware.c

```diff
@@ -35,8 +28,15 @@ int main() {
 
     for (;;) {
         ice_usb_task();
-        ice_smem_write_async(ICE_SSRAM_SPI_CS_PIN, START_ADDR, write_data, sizeof(write_data));
-        ice_smem_read_async(ICE_SSRAM_SPI_CS_PIN, read_data, START_ADDR, sizeof(read_data));
+
+        // The write callback chains a read sequence immediately after the write sequence from an interrupt handler,
+        // without any interaction with the main thread.
+        void write_callback(void* context) {
+            ice_smem_read_async(ICE_SSRAM_SPI_CS_PIN, read_data, START_ADDR, sizeof(read_data), NULL, NULL);
+        }
```

Comment:
Is that a nested function? Would that not decrease portability?

Reply:
Done